src/modal.js

```diff
@@ -65,7 +64,7 @@ class NaughtyModal extends Component {
           />
         </DialogContent>
         <DialogActions>
-          <Button disabled={prompt !== input} onClick={onClose}>
+          <Button disabled={ prompt === originalPrompt } onClick={onClose}>
```

Comment:
### Bug
As the user updates their input, the confirmation should remain disabled until their input matches the dynamically updating prompt.

#### Expected Behaviour
Until such a time that the content of the text field matches the prompt string, the confirmation button should remain disabled.

#### Current Behaviour
The confirmation button is disabled when the text field input is empty, but is unlocked once input is given by the user, irrespective of whether it matches the prompt or not.

Reply:
Thank you for your feedback, Nick. I can fix that behaviour; I honestly didn't realise that's what it was meant to do. Perhaps you should document this. Regarding the semicolons, I'll take them out, but I recommend you reconfigure your linter to require semicolons.

Reply:
You're right, I will update the guidelines of this project to better convey the proper behaviour.

Regarding semicolon use, I do not personally feel their use is significantly warranted, and so that is the configuration I apply for my personal projects. If I work on projects that use semicolons, I will do my best to adhere to their standards and the configuration they use. The difference to me seems trivial, for as far as I can tell it's an argument over a single boolean in a configuration file, and thus the best approach is to follow the lead of others when working on their projects.

*Avoid conflict where possible, especially merge conflicts!*